src/components/MyObservations/ToolbarContainer.js

```diff
@@ -112,7 +112,7 @@ const ToolbarContainer = ( {
     if ( uploadInProgress ) {
       const translationParams = {
         total: numToUpload,
-        currentUploadCount: numFinishedUploads + 1
+        currentUploadCount: Math.min( numFinishedUploads + 1, numToUpload )
```

Comment:
At least in tests I was seeing some cases of "Uploading 3 of 2 observations" so this should prevent that... I hope.

Reply:
Okay, I did not see this particular problem in testing but I can imagine having a race condition where `numFinishedUpload` ticks one higher before `uploadInProgress` is false.